### PR TITLE
monasca: Log agent config

### PR DESCRIPTION
**Pull request by matrixik**

### CODE REVIEW

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 1]
```diff
@@ -20,9 +20,12 @@
 keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
 
 monasca_server = node_search_with_cache("roles:monasca-server").first
+if monasca_server.nil?
+  Chef::Log.warn("No monasca-server found.")
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: nit: could you add more details? like ```No monasca-server found. Skip monasca-log-agent setup```

File: chef/cookbooks/monasca/recipes/metric_agent.rb [hunk 2]
```diff
@@ -21,8 +21,11 @@
 
 keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
 
-monasca_servers = search(:node, "roles:monasca-server") || []
-monasca_server = monasca_servers[0]
+monasca_server = node_search_with_cache("roles:monasca-server").first
+if monasca_server.nil?
+  Chef::Log.warn("No monasca-server found.")
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/metric_agent.rb]: nit: more details (same as for log-agent)

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 3]
```diff
@@ -31,6 +34,15 @@
   "/var/log/zypper.log" => "system"
 }
 
+ruby_block "find log files" do
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: Just for my understanding - With that list you tell monasca-log-agent which files you want to transfer to monasca-log-api ?

Comment (samirjorina) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: Yes, that's correct. This block of code detects what log files are available and these files will be monitored. Consequently, log entries from these files will be sent to monasca-log-api.

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 4]
```diff
@@ -31,6 +34,15 @@
...
+  block do
+    log_dirs =
+      Dir.entries("/var/log")
+         .select { |e| File.directory?("/var/log/#{e}") && !(e == "." || e == "..") }
+    log_dirs.each { |d| log_files["/var/log/#{d}/**/*.log"] = d.downcase }
```
Comment (dirkmueller) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: I guess we need to skip monasca related log files here..?

Comment (matrixik) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: Done

**Comment by dirkmueller** [comment]
regarding to the delete of agent.yaml.erb - it is correct that it is unused currently, but it feels like we should be switching to that method instead since monasca_setup is very slow and very buggy right now.

File: chef/cookbooks/monasca/recipes/server.rb [hunk 5]
```diff
@@ -11,6 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/server.rb]: Unrelated to the commit msg

File: crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml [hunk 6]
```diff
@@ -3,6 +3,7 @@
     = header show_raw_deployment?, true
 
   .panel-body
+
```
Comment (toabctl) [inline crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml]: unrelated to the commit msg

File: crowbar_framework/app/models/monasca_service.rb [hunk 7]
```diff
@@ -93,17 +93,17 @@ def create_proposal
     nodes = NodeObject.all
 
     monasca_server = select_nodes_for_role(nodes, "monasca-server", "monitoring") || []
-    log_agent_nodes = select_nodes_for_role(nodes, "monasca-log-agent", "compute") || []
     agent_nodes = select_nodes_for_role(nodes, "monasca-agent", "compute") || []
+    log_agent_nodes = select_nodes_for_role(nodes, "monasca-log-agent", "compute") || []
```
Comment (toabctl) [inline crowbar_framework/app/models/monasca_service.rb]: why that change?

**Comment by toabctl** [comment]
An alternative which uses chef to configure the basic plugins and the agent itself: https://github.com/crowbar/crowbar-openstack/pull/928

File: crowbar_framework/app/models/monasca_service.rb (hunk 7 above)
Comment (matrixik) [inline crowbar_framework/app/models/monasca_service.rb]: To have all of them in same order like e.g in `default.rb`, for easier top-down code reading

**Comment by matrixik** [comment]
Reworked

File: crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml (hunk 6 above)
Comment (matrixik) [inline crowbar_framework/app/views/barclamp/monasca/_edit_attributes.html.haml]: Moved to separate commit

**Comment by matrixik** [comment]
And rebased the second time...

**Comment by toabctl** [comment]
triggered the CI. Sorry for being so unresponsive...

File: chef/cookbooks/monasca/recipes/log_agent.rb [hunk 8]
```diff
@@ -26,7 +26,6 @@
 end
 
 log_agent_dimensions = {
-  service: "monitoring",
```
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: This is unrelated to the commit msg.

File: chef/cookbooks/monasca/recipes/log_agent.rb (hunk 4 above)
Comment (toabctl) [inline chef/cookbooks/monasca/recipes/log_agent.rb]: Can you please squash that into the commit where you introduce this?

**Comment by toabctl** [comment]
I addressed all comments in https://github.com/crowbar/crowbar-openstack/pull/941
